### PR TITLE
implement MonadRec for Function

### LINKED ISSUE #17: Make Function instance of MonadRec
In [sanctuary-type-classes](https://github.com/sanctuary-js/sanctuary-type-classes/blob/master/index.js#L790) we have made function instance of MonadRec so it could be implemented here too.

### PR DESCRIPTION
**Pull request by safareli**
fix #17

### CODE REVIEW

File: test/Test/Main.purs [hunk 1]
```diff
@@ -25,6 +25,12 @@ loop n = tailRecM go n
     pure (Done unit)
   go n' = pure (Loop (n' - 1))
 
+loopFunction :: Int -> ({result :: Int, step :: Int} -> Int)
```
Comment (safareli) [inline test/Test/Main.purs:28]: Parens could be omitted here just left for conveniens

**Comment by safareli** [comment]
After this Trampoline from ps-free could be implemented using just function
```diff
-type Trampoline = Free Lazy
+type Trampoline = Free ((->) Unit) 

done :: forall a. a -> Trampoline a
done = pure

suspend :: forall a. Trampoline a -> Trampoline a
suspend = suspendF

-delay' :: forall a. Lazy a -> Trampoline a
-delay' = liftF

delay :: forall a. (Unit -> a) -> Trampoline a
-delay = delay' <<< defer
+delay = liftF

runTrampoline :: forall a. Trampoline a -> a
-runTrampoline = runFree force
+runTrampoline = runFree \f -> f Unit
```

**Comment by safareli** [comment]
@paf31 ping

**Comment by paf31** [comment]
Thanks!

**Pull request merged**